imod/mf6/drn.py

```diff
@@ -155,3 +169,136 @@ def _validate(self, schemata, **kwargs):
 
     def get_regrid_methods(self) -> Optional[dict[str, Tuple[RegridderType, str]]]:
         return self._regrid_method
+
+    @classmethod
+    def from_imod5_data(
+        cls,
+        imod5_data: dict[str, dict[str, GridDataArray]],
+        target_discretization: StructuredDiscretization,
+        target_npf: NodePropertyFlow,
+        regridder_types: Optional[
+            dict[str, dict[str, tuple[RegridderType, str]]]
+        ] = None,
+        allocation_methods: Optional[dict[str, ALLOCATION_OPTION]] = None,
+        distributing_option: Optional[dict[str, DISTRIBUTING_OPTION]] = None,
+    ) -> list["Drainage"]:
+        """
+        Construct a drainage-package from iMOD5 data, loaded with the
+        :func:`imod.formats.prj.open_projectfile_data` function.
+
+        .. note::
+
+            The method expects the iMOD5 model to be fully 3D, not quasi-3D.
+
+        Parameters
+        ----------
+        imod5_data: dict
+            Dictionary with iMOD5 data. This can be constructed from the
+            :func:`imod.formats.prj.open_projectfile_data` method.
+        target_grid: GridDataArray
+            The grid that should be used for the new package. Does not
+            need to be identical to one of the input grids.
+        regridder_types: dict, optional
+            Optional dictionary with regridder types for a specific variable.
+            Use this to override default regridding methods.
+        regridder_types:  dict[str, dict[str, tuple[RegridderType, str]]], optional
+            regridder types per array, per package. The first key is the package name
+            as present imod 5 and the second one the input array
+        allocation_methods: dict[str, ALLOCATION_OPTION]], optional
+            allocation option per package. The key is the package name as present in
+            imod 5. All or none should be present.
+        distributing_option: dict[str, DISTRIBUTING_OPTION]], optional
+            distributing option option per package. The key is the package name as present in
+            imod 5. All or none should be present.
+
+        Returns
+        -------
+        A list of Modflow 6 Drainage packages.
+        """
+
+        drainage_keys = [k for k in imod5_data.keys() if k[0:3] == "drn"]
+        if allocation_methods is None:
+            allocation_methods = dict.fromkeys(
+                drainage_keys, ALLOCATION_OPTION.first_active_to_elevation
+            )
+        if distributing_option is None:
+            distributing_option = dict.fromkeys(
+                drainage_keys, DISTRIBUTING_OPTION.by_corrected_transmissivity
+            )
+        if regridder_types is None:
+            regridder_types = dict.fromkeys(drainage_keys, Drainage._regrid_method)
```

Comment:
These dictionaries won't be necessary anymore for a single package, this will be moved to the Modflow6Simulation object. Best is to follow the namedtuple/dataclass idea proposed [here](https://github.com/Deltares/imod-python/issues/960) there. Something like:

```python
@dataclass
class DefaultSimulationAllocationOptions:
      drn: ALLOCATION_OPTION = ALLOCATION_OPTION.first_active_to_elevation
      riv: ALLOCATION_OPTION = ALLOCATION_OPTION.stage_to_elevation
```

We can use pydantic dataclasses to do conduct input validation. https://docs.pydantic.dev/latest/concepts/dataclasses/

Reply:
as discussed offline, this proposal will end up in the simulation import PR.